### PR TITLE
docs[patch]: Adds components for prereqs, compatibility, fix chat model tab issue

### PR DESCRIPTION
**Pull request by jacoblee93**
Added to `docs/how_to/tools_runtime` as a proof of concept, will apply everywhere if we like.

A bit more compact than the default callouts, will help standardize the layout of our pages since we frequently use these boxes.

Screenshot 2024-07-23 at 4 49 02 PM

### CODE REVIEW

**Comment by eyurtsev** [comment]
@jacoblee93 is there any way to reduce the API we need to know about inside the notebooks? And add a directive that'll get post-processed into the appropriate format?

File: docs/src/theme/Prerequisites.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import Admonition from '@theme/Admonition';
+
+export default function Prerequisites({ titlesAndLinks }) {
+    return (
+        <Admonition type="info" title="Recommended Prereading" icon="📚">
```
Comment (baskaryan) [inline docs/src/theme/Prerequisites.js:6]: i think i like "Conceptual Prereqs" more than recommended reading — they really are prereqs for understanding what's going on, not just suggestions

Comment (jacoblee93) [reply docs/src/theme/Prerequisites.js]: Changed back to `Prerequisites`

**Comment by jacoblee93** [comment]
> @jacoblee93 is there any way to reduce the API we need to know about inside the notebooks? And add a directive that'll get post-processed into the appropriate format?

We *could* swizzle the admonitions component and use a custom one if you pass a specific type in. However, I think part of the idea here is actually to reduce the amount of customization a writer can add here, and if we go with that approach you'd be able to enter whatever you want.

**Review by baskaryan — APPROVED** [review]

**Review by efriis — APPROVED** [review]
I like it! Should we do the same format for related section in #24829 ?

**Comment by jacoblee93** [comment]
We could! I think fine as a followup though

**Comment by daniel-jscraft** [comment]
ha! this is nice 👍

**Pull request merged**